aws/rust-runtime/aws-config/src/imds/client.rs

```diff
@@ -251,6 +252,9 @@ pub enum ImdsError {
     /// An error occurred communication with IMDS
     IoError(Box<dyn Error + Send + Sync + 'static>),
```

Comment:
I think in the case of IMDS, I'd probably just put this into IOError instead of adding another variant